### ISSUE TITLE
can't create sky from an EquirectMap without a warning

### ISSUE THREAD

**Issue by stephengold** [jMonkeyEngine/jmonkeyengine#1414]
Seen in Jayfella's jme-vehicles project:
```text
Oct 23, 2020 5:37:55 PM com.jme3.material.Material checkSetParam
WARNING: Material parameter being set: Texture with type Texture2D doesn't match definition types TextureCubeMap
```
This diagnostic is triggered by
```java
SkyFactory.createSky(assetManager, "Textures/Sky/quarry_03_4k.jpg", SkyFactory.EnvMapType.EquirectMap);
```
in Main.java.

The resulting render looks great; I just want to eliminate the diagnostic message. I tried to work around the warning without modifying either jme3-core or the JPEG, but was unsuccessful.

Sky.j3md has
```j3md
    MaterialParameters {
        TextureCubeMap Texture
```

But I can't use "quarry_03_4k.jpg" as a `TextureCubeMap` because it's 4096x2048 (not square).

This is one case where type-checking material parameters is awkward.

**Comment by tlf30** [comment]
I have seen the same issue in the past. It would be nice to 'disable' the error message. (AKA, remove the check)

**Comment by riccardobl** [comment]
I think there is a problem in the design of the material here.

The matdef wants a TextureCubeMap that should always be mapped to a samplerCube, passing anything that is not a cubemap there would be wrong by definition, this works because the Sky shader uses a macro to define Texture as a sampler2D if EQUIRECT_MAP or SPHERE_MAP are set.

Imo this is bad practice, the matdef should have two params, one for the cubemap and one for the texture2D, or be divided in two different materials.

**Comment by stephengold** [comment]
I like the solution of 2 different material definitions for sky.

**Comment by tlf30** [comment]
The sane warning occures when loading a hdr file as well.

**Comment by stephengold** [comment]
Solved in "master"branch at d7a363c .
